src/ucp/rndv/proto_rndv.c

```diff
@@ -227,19 +227,17 @@ ucp_proto_rndv_ctrl_perf(const ucp_proto_init_params_t *params,
 
 static ucs_status_t
 ucp_proto_rndv_ctrl_init_priv(const ucp_proto_rndv_ctrl_init_params_t *params,
-                              ucp_lane_index_t lane)
+                              ucp_proto_rndv_ctrl_priv_t *rpriv)
 {
     const ucp_proto_init_params_t *init_params = &params->super.super;
     ucp_context_h context                      = init_params->worker->context;
-    ucp_proto_rndv_ctrl_priv_t *rpriv          = init_params->priv;
     const ucp_proto_select_param_t *select_param;
     ucp_proto_select_param_t remote_select_param;
     ucp_memory_info_t mem_info;
     uint32_t op_attr_mask;
 
-    select_param            = init_params->select_param;
-    *init_params->priv_size = sizeof(ucp_proto_rndv_ctrl_priv_t);
-    rpriv->lane             = lane;
+    select_param = init_params->select_param;
+    rpriv->lane  = params->lane;
```

Comment:
```suggestion
    rpriv->lane  = ucp_proto_rndv_find_ctrl_lane(init_params);
    if (lane == UCP_NULL_LANE) {
        return UCS_ERR_UNSUPPORTED;
    }
```

If I understand the things correctly this change would allow to remove `lane` field from `ucp_proto_rndv_ctrl_init_params_t` and this check too (https://github.com/openucx/ucx/pull/9755/files#diff-a00876d90c38d621fb6da5f5ccb23dddf8648c77e9967273f59505ad8a623248R300)

Reply:
it's needed here: https://github.com/openucx/ucx/blob/d9d9c0c1a6aeb7f87b837a958c227a3c20b5f441/src/ucp/tag/offload/rndv.c#L183